service/builder/pipelines_builder.go

```diff
@@ -60,15 +61,20 @@ func (bps BuiltPipelines) StartProcessors(logger *zap.Logger, host component.Hos
 }
 
 func (bps BuiltPipelines) ShutdownProcessors(logger *zap.Logger) error {
+	var errors []error
 	for cfg, bp := range bps {
 		logger.Info("Pipeline is shutting down...", zap.String("pipeline", cfg.Name))
 		for _, p := range bp.processors {
 			if err := p.Shutdown(); err != nil {
-				return err
+				errors = append(errors, err)
 			}
 		}
 		logger.Info("Pipeline is shutdown.", zap.String("pipeline", cfg.Name))
 	}
+
+	if len(errors) != 0 {
+		return multierr.Combine(errors...)
```

Comment:
See https://pkg.go.dev/go.uber.org/multierr?tab=doc#Combine, it will produce error message like this:
```
the following errors occurred:
 -  call 1 failed
 -  call 3 failed
 -  call 5 failed
```
It may be worth wrapping errors to provide more details (e.g. pipeline name).

Reply:
Knowing which pipeline failed with which error message is very valuable.

Reply:
Unfortunately, `Processor` interface doesn't provide a name or other identifying information, so that would need to be addressed separately.

Reply:
Replaced with `oterr.CombineErrors`

Reply:
I thought there was a pipeline name a few lines above?

```go
logger.Info("Pipeline is shutting down...", zap.String("pipeline", cfg.Name))
```

Reply:
`cfg.Name` might actually be good enough for now, though there are multiple processors per pipeline.